Added cli-friendly updates

smcdon opened the pull request:
-- Output is written to STDOUT by default. Supply a file path to write
to file.
-- Added --silent option for easy piping to other utilities.

Other minor changes:
-- Simplify the config passed to the library by removing duplicate hash and changing the appropriate keys.

smcdon committed:
Added cli-friendly updates

-- Output is written to STDOUT by default. Supply a file path to write
to file.
-- Added --silent option for easy piping to other utilities.
-- Added --max option to limit the number of results. Good for
testing, especially when writing to STDOUT.

smcdon committed:
Added cli-friendly updates

-- Output is written to STDOUT by default. Supply a file path to write
to file.
-- Added --silent option for easy piping to other utilities.

robbydyer commented:
I haven't been actively developing on this - @smcdon I trust you have tested these changes and everything works ok?

smcdon committed:
Merge branch 'cli-updates' of github.com:smcdon/stash-query into cli-updates

smcdon committed:
Merge branch 'cli-updates' of github.com:smcdon/stash-query into cli-updates

smcdon committed:
Merge branch 'cli-updates' of github.com:smcdon/stash-query into cli-updates

smcdon commented:
Hi @robbydyer,

I found a bug in my implementation. I've fixed it, and also added another feature as well, but to simplify the commit history I'm going to checkout a new branch with the changes and create a new pull request.
